src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -154,27 +158,39 @@ def before_request(
         limiter = asyncio.Semaphore(concurrency_level)
 
         pdf = PdfReader(io.BytesIO(file.content))
+
+        page_range_start, page_range_end = form_utils.get_page_range(
+            form_data,
+            key=PARTITION_FORM_PAGE_RANGE_KEY,
+            max_pages=len(pdf.pages),
+        )
+
+        page_count = min(len(pdf.pages), page_range_end - page_range_start + 1)
+        logger.info(
+            "Splitting pages %d to %d (%d total)",
+            page_range_start,
+            page_range_end,
+            page_count,
+        )
```

Comment:
Is `min` necesarry here? What I mean is, if the `page_range_end - page_range_start + 1` is higher than `len(pdf.pages)` then the info log will be wrong because the pages range won't match page count.

Also `get_page_range` will throw an error if it's out of bounds so is there a case when `page_count` will be assigned `len(pdf.pages)`?

I might be missing something here so correct me if I'm wrong.

Reply:
Good catch, we don't need the min since `end - start + 1` will be the correct value. The boundary checks in `get_page_range` ensure that `start` and `end` are less then or equal to `max_pages`, so at most this calculation will be `max_pages - 1 + 1`, or `len(pdf.pages)`.